src/exporter/DaemonMetricCollector.cc

```diff
@@ -82,35 +82,41 @@ std::string boost_string_to_std(boost::json::string js) {
 
 std::string quote(std::string value) { return "\"" + value + "\""; }
 
-void DaemonMetricCollector::dump_asok_metrics() {
+void DaemonMetricCollector::dump_asok_metrics(bool sort_metrics, int64_t counter_prio,
+                                              bool sockClientsPing, std::string dump_response,
+                                              std::string schema_response,
+                                              bool config_show_response) {
   BlockTimer timer(__FILE__, __FUNCTION__);
 
   std::vector<std::pair<std::string, int>> daemon_pids;
 
   int failures = 0;
-  bool sort = g_conf().get_val<bool>("exporter_sort_metrics");
+  bool sort;
+  sort = sort_metrics ? true : g_conf().get_val<bool>("exporter_sort_metrics");
```

Comment:
There is a lot of ternary operations for these things. Why not just extract get_val from this function and move it to `request_loop` so we don't have this many ternaries.

Reply:
I've done this to make it easier for unit tests as well as they call this function for exporting metrics.

Reply:
I understand what you did there. I'm just saying that since we are using parameters for `dump_asok_metrics`, might be worth to move generation of those parameters out of it so that the logic is not too hacky.